### PR TITLE
[master-2.x] Force sync state root from all peers

### PR DESCRIPTION
**Pull request by KickSeason**
Fix root sync slowly in `mast-2.x`, especially many peers are `2.10.3`.

 * Treat state root sync task as global task
 * Ask all connected peers when get state roots

 > Risk: peers will do repeated work and node will receive some redundant messages from different peers.

### CODE REVIEW

**Comment by KickSeason** [comment]
@erikzhang @shargon Can you review this?

File: neo/Network/P2P/TaskManager.cs [hunk 1]
```diff
@@ -281,15 +286,11 @@ private void RequestTasks(TaskSession session)
                 var height = Blockchain.Singleton.Height;
                 if (state_height + 1 < height)
                 {
-                    var state = Blockchain.Singleton.GetStateRoot((uint)(state_height + 1));
-                    if (state is null || state.Flag == StateRootVerifyFlag.Unverified)
```
Comment (shargon) [inline neo/Network/P2P/TaskManager.cs:285]: Why remove this check?

Comment (KickSeason) [reply neo/Network/P2P/TaskManager.cs:285]: After every block persisted,  local state root will be calculated and persist into `StateRoots` with `Unverified` flag. Only check the block height is enough.
https://github.com/neo-project/neo/blob/95596f93a604f48d0a962b092b143cbdf0769bf5/neo/Ledger/Blockchain.cs#L656

**Review by shargon — APPROVED** [review]

File: neo/Network/P2P/TaskManager.cs [hunk 2]
```diff
@@ -58,7 +59,7 @@ private void OnInternalTaskCompleted(UInt256 hash)
         {
             if (!sessions.TryGetValue(Sender, out TaskSession session))
                 return;
-            session.Tasks.Remove(hash);
+            if (hash != StateRootTaskHash) session.Tasks.Remove(hash);
```
Comment (erikzhang) [inline neo/Network/P2P/TaskManager.cs:62]: What's the difference?

Comment (KickSeason) [reply neo/Network/P2P/TaskManager.cs:62]: `StateRootTaskHash` only exits in `GlobalTasks`.

Comment (erikzhang) [reply neo/Network/P2P/TaskManager.cs:62]: But if it doesn't exist, `Remove()` will do nothing. So we don't need to check whether it is `StateRootTaskHash`.

Comment (KickSeason) [reply neo/Network/P2P/TaskManager.cs]: done

**Review by shargon — APPROVED** [review]

**Comment by superboyiii** [comment]
I've already applied it on testnet seeds and all works well.

**Pull request merged**